aidial_assistant/model/model_client.py

```diff
@@ -70,43 +76,53 @@ async def agenerate(
         extra_results_callback: ExtraResultsCallback | None = None,
         **kwargs,
     ) -> AsyncIterator[str]:
-        async with ClientSession(read_bufsize=self.buffer_size) as session:
-            openai.aiosession.set(session)
-
-            model_result = await openai.ChatCompletion.acreate(
-                messages=[message.to_openai_message() for message in messages],
-                **self.model_args | kwargs,
-            )
-
-            finish_reason_length = False
-            async for chunk in model_result:  # type: ignore
-                usage: Usage | None = chunk.get("usage")
-                if usage:
-                    prompt_tokens = usage["prompt_tokens"]
-                    self._total_prompt_tokens += prompt_tokens
-                    self._total_completion_tokens += usage["completion_tokens"]
-                    if extra_results_callback:
-                        extra_results_callback.on_prompt_tokens(prompt_tokens)
-
+        model_result = await self.client.chat.completions.create(
+            **self.model_args,
+            extra_body=kwargs,
+            stream=True,
+            messages=[message.to_openai_message() for message in messages],  # type: ignore
```

Comment:
I would rather see type-cast, then type ignore:
```
cast(
                List[ChatCompletionMessageParam],
                [message.to_openai_message() for message in messages],
            )
```
Or even better: use the types which are compatible with `ChatCompletionMessageParam` in the first place to avoid the casting.

Reply:
No longer needed.